source/module_io/read_cube.cpp

```diff
@@ -147,43 +140,51 @@ void ModuleIO::trilinear_interpolate(
     delete[] read_rho;
 }
 
-ModuleIO::CubeInfo ModuleIO::read_cube(const std::string& file)
+bool ModuleIO::read_cube(const std::string& file,
+    std::vector<std::string>& comment,
+    int& natom,
+    std::vector<double>& cell_pos,
+    std::vector<int>& nvoxel,
+    std::vector<std::vector<double>>& axis_vecs,
```

Comment:
Good. There are additional minor comments:
1. it is not cell_pos, instead it is the coordinate of "(0, 0, 0)" where volumetric data will span. So a more appropriate name would be `origin` or something
2. it is okay to use std::vector<int> to store number of voxels in three direction, but my personal idea is the dimensions of cube file is well-defined to be 3, so I prone expose nx, ny and nz instead of storing them in a length-mutable container
3. it is not axis_vecs, it is something like delta_x, delta_y and delta_z, their mixed product will be the volume of an voxel.

Reply:
http://www.paulbourke.net/dataformats/cube/